### ISSUE TITLE
Crashes when file path contains colon (`:`)

### ISSUE THREAD

**Issue by ribose-jeffreylau** [metanorma/mn2pdf#120]
## Trace

<details>

<summary>click</summary>

```console

> pwd
/Users/username/share/src/test:
> bundle exec metanorma -x pdf main.adoc
...
<snip>
java -Xss5m -Xmx2048m -Dapple.awt.UIElement=true -jar /Users/username/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/mn2pdf-1.41.1/lib/../bin/mn2pdf.jar --xml-file "/Users/username/share/src/test:/main.presentation.xml" --xsl-file "/Users/username/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/metanorma-ribose-2.0.8/lib/isodoc/ribose/ribose.standard.xsl" --pdf-file "/Users/username/share/src/test:/main.pdf" --syntax-highlight  --font-manifest "/var/folders/v7/bpr811_x73bfcjbzg8mvb_v40000gq/T/fontist_locations20220428-96816-linecn.yml"
[mn2pdf] Fatal: mn2pdf Preparing...
Input: XML (/Users/username/share/src/test:/main.presentation.xml)
Input: XSL (/Users/username/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/metanorma-ribose-2.0.8/lib/isodoc/ribose/ribose.standard.xsl)
Output: PDF (/Users/username/share/src/test:/main.pdf)

[INFO] XSL-FO file preparation...
(Location of error unknown)java.io.FileNotFoundException: /Users/username/share/src/tes/Users/username/share/src/test:/t:/main.presentation.xml (No such file or directory)
Transforming to PDF...
Default page-height set to: 11.00in
Default page-width set to: 8.50in
javax.xml.transform.TransformerException: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 1; Premature end of file.
	at org.apache.xalan.transformer.TransformerIdentityImpl.transform(TransformerIdentityImpl.java:502)
	at org.metanorma.fop.PDFGenerator.runFOP(PDFGenerator.java:509)
	at org.metanorma.fop.PDFGenerator.convertmn2pdf(PDFGenerator.java:394)
	at org.metanorma.fop.PDFGenerator.process(PDFGenerator.java:272)
	at org.metanorma.fop.mn2pdf.main(mn2pdf.java:334)
Caused by: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 1; Premature end of file.
	at java.xml/com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:204)
	at java.xml/com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.fatalError(ErrorHandlerWrapper.java:178)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:400)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:327)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLScanner.reportFatalError(XMLScanner.java:1471)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(XMLDocumentScannerImpl.java:1013)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:112)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:542)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:889)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:825)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1224)
	at org.apache.xalan.transformer.TransformerIdentityImpl.transform(TransformerIdentityImpl.java:485)
	... 4 more
---------
org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 1; Premature end of file.
	at java.xml/com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:204)
	at java.xml/com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.fatalError(ErrorHandlerWrapper.java:178)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:400)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:327)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLScanner.reportFatalError(XMLScanner.java:1471)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(XMLDocumentScannerImpl.java:1013)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:112)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:542)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:889)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:825)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1224)
	at org.apache.xalan.transformer.TransformerIdentityImpl.transform(TransformerIdentityImpl.java:485)
	at org.metanorma.fop.PDFGenerator.runFOP(PDFGenerator.java:509)
	at org.metanorma.fop.PDFGenerator.convertmn2pdf(PDFGenerator.java:394)
	at org.metanorma.fop.PDFGenerator.process(PDFGenerator.java:272)
	at org.metanorma.fop.mn2pdf.main(mn2pdf.java:334)
make: *** [main.html] Error 1

```
</details>

## Line of interest

... is probably this line here from the trace:

```
(Location of error unknown)java.io.FileNotFoundException: /Users/username/share/src/tes/Users/username/share/src/test:/t:/main.presentation.xml (No such file or directory)
Transforming to PDF...
```

Notice how it's trying to read from a jumbled path.

**Comment by ronaldtse** [comment]
@Intelligent2013 I think we want to ensure that we are only reading the file once?

```java
        this.documentFilePath = this.fXML.getParent();
```

**Comment by ribose-jeffreylau** [comment]
UPDATE: The trace should be from version `1.41.1`.

**Comment by Intelligent2013** [comment]
For input xml: `/home/intluser/src/abc:/rice-en.final.presentation.xml`
- `fXML.getParent()` return `/home/intluser/src/abc:` - correct
- `(Source) src.getSystemId()` returns  `file:/home/intluser/src/abc:/rice-en.final.presentation.xml` correct.
- errors occurs in `transformerFO.transform(src, sr);`:
`(Location of error unknown)java.io.FileNotFoundException: /home/intluser/src/ab/home/intluser/mn2pdf/c:/rice-en.final.presentation.xml (No such file or directory)`

javax.xml.transform.Transformer splits the path `/home/intluser/src/abc:/rice-en.final.presentation.xml` into two parts:
- `/home/intluser/src/ab`
- `c:/rice-en.final.presentation.xml`
and put current path `/home/intluser/mn2pdf` between them.

I'll try fix by passing object Source created from String, not File.

**Comment by Intelligent2013** [comment]
Issue fixed. Updated mn2pdf will be available in next release. I'll post here about it.

**Comment by Intelligent2013** [comment]
Sorry for a long answer. The issue fixed in mn2pdf v1.44.
